runtime/databricks/automl_runtime/forecast/prophet/diagnostics.py

```diff
@@ -0,0 +1,163 @@
+#
+# Copyright (C) 2022 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from __future__ import absolute_import, division, print_function
+from prophet.diagnostics import single_cutoff_forecast
+
+import logging
+from tqdm.auto import tqdm
+import concurrent.futures
+
+import pandas as pd
+
+
+logger = logging.getLogger('prophet')
+
+def cross_validation(model, horizon, period=None, initial=None, parallel=None, cutoffs=None, disable_tqdm=False):
+    """Cross-Validation for time series.
+
+    This function is same as prophet's native function
+    https://github.com/facebook/prophet/blob/main/python/prophet/diagnostics.py#L61
+    except that we support user to pass in horizon as pandas.DateOffset
+    object directly, this would support the non-daily horizon forecast,
```

Comment:
nit: we previously already support some non-daily frequency, e.g. minutes, seconds. Probably instead of saying support non-daily horizon forecast, we can directly say support monthly, quarterly and annually forecast, and mention it was the constraint from `pd.Timedelta` that it doesn't support month and year as unit.